src/analysis/enums.rs

```diff
@@ -84,6 +85,38 @@ pub fn new(env: &Env, obj: &GObject, imports: &mut Imports) -> Option<Info> {
         None,
         None,
     );
+
+    // Gir does not currently mark the first parameter of associated enum functions -
+    // that are identical to its enum type - as instance parameter since most languages
+    // do not support this.
+    for f in &mut functions {
+        if f.parameters.c_parameters.is_empty() {
+            continue;
+        }
+
+        let first_param = &mut f.parameters.c_parameters[0];
+
+        if first_param.typ == enumeration_tid {
+            first_param.instance_parameter = true;
+
+            let t = f
+                .parameters
+                .transformations
+                .iter_mut()
+                .find(|t| t.ind_c == 0)
+                .unwrap();
+
+            if let TransformationType::ToGlibScalar { name, .. } = &mut t.transformation_type {
+                *name = "self".to_owned();
+            } else {
+                panic!(
+                    "Enum function instance param must be passed as scalar, not {:?}",
+                    t.transformation_type
```

Comment:
Wouldn't this panic if there was a non-instance enum function? Something that does not take `Self` as first parameter but something completely different?

Reply:
If it takes something different than `Self` it should not reach past `if first_param.typ == enumeration_tid {` correct?

Reply:
Indeed, sorry :)